src/olympia/addons/serializers.py

```diff
@@ -309,7 +309,8 @@ def to_representation(self, obj):
         if ('request' in self.context and
                 'wrap_outgoing_links' in self.context['request'].GET):
             for key in ('homepage', 'support_url', 'contributions_url'):
-                data[key] = self.outgoingify(data[key])
+                if key in data:
```

Comment:
This seems fine but there is a risk that you'll miss a field later on if it gets renamed. Another way to do it would be to use a serializer specific mapping:

```python
# In addons/serializers.py

class AddonSerializer(serializers.ModelSerializer):
    class Meta:
        _external_url_fields = ('homepage', 'support_url', 'contributions_url')

# In discovery/serializers.py

class DiscoveryAddonSerializer(AddonSerializer):
    class Meta:
        _external_url_fields = []

```

However, I don't know if it's worth it or not.

Reply:
I'm didn't want to go this route because that would require you to explicitly set `_external_url_fields` in every serializer inheriting from `AddonSerializer` which does not need the fields in question. Ultimately IMHO those serializers should not have to care about this, they are not including the fields so it should just ignore them.

Reply:
WFM